transport/src/main/java/io/netty/channel/DefaultChannelId.java

```diff
@@ -296,19 +305,37 @@ public int compareTo(final ChannelId o) {
         if (o instanceof DefaultChannelId) {
             // lexicographic comparison
             final byte[] otherData = ((DefaultChannelId) o).data;
-            int len1 = data.length;
-            int len2 = otherData.length;
-            int len = Math.min(len1, len2);
+            final int otherLen = otherData.length;
+            final byte[] data = this.data;
+            final int len = data.length;
+            if (len == otherLen) { // when machineIdLen is the same
+                for (int i = 0; i < len; ++i) {
+                    final byte x = data[i];
+                    final byte y = otherData[i];
+                    if (x != y) {
+                        return (x & 0xff) - (y & 0xff);
+                    }
+                }
+                return 0;
+            }
 
-            for (int k = 0; k < len; k++) {
-                byte x = data[k];
-                byte y = otherData[k];
+            // when machineIdLen is different
+            final int nonMachineIdLen = PROCESS_ID_LEN + SEQUENCE_LEN + TIMESTAMP_LEN + RANDOM_LEN;
+            final int machineIdLen = len - nonMachineIdLen;
+            final int otherMachineIdLen = otherLen - nonMachineIdLen;
+            final int minMachineIdLen = Math.min(machineIdLen, otherMachineIdLen);
+            assert machineIdLen != otherMachineIdLen;
```

Comment:
I think this is over-thinking it. As long as the sort order is consistent, it should be fine. No need to make it match the long-text sorting, IMO.

Reply:
Sure. Let me fix it